File: DSA/RATNA_KHANNA_590013141/DAY_38_QUESTION_1.c
#include <stdio.h>
#include <stdlib.h>

void nextGreaterElement(int arr[], int n) {
    int result[n];     
    int stack[n];      
    int top = -1;    

    for (int i = 0; i < n; i++) {
        while (top != -1 && arr[i] > arr[stack[top]]) {
            int index = stack[top--];     
            result[index] = arr[i];      
        }
        stack[++top] = i;  
    }

    while (top != -1) {
        result[stack[top--]] = -1;
    }

    printf("Next Greater Elements:\n");
    for (int i = 0; i < n; i++) {
        printf("%d -> %d\n", arr[i], result[i]);
    }
}

int main() {
    int n;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    nextGreaterElement(arr, n);

    return 0;
}

File: DSA/Pranvkumar_590011587/Day38_Question1.py
def next_greater_element(arr):
    n = len(arr)
    res = [-1] * n
    stack = []
    for i in range(n - 1, -1, -1):
        while stack and stack[-1] <= arr[i]:
            stack.pop()
        res[i] = stack[-1] if stack else -1
        stack.append(arr[i])
    return res

# Examples
print(next_greater_element([4, 5, 2, 25]))   
print(next_greater_element([13, 7, 6, 12]))  